test/Spago/Build.purs

```diff
@@ -102,36 +104,40 @@ spec = Spec.around withTempDir do
 
       Spec.describe "fails when there are imports from transitive dependencies and" do
         let
-          setupSrcTransitiveTests spago = do
+          setupSrcTransitiveTests spago installConsoleAndEffect = do
             spago [ "init", "--name", "7368613235362d34312f4e59746b7869335477336d33414d72" ] >>= shouldBeSuccess
             spago [ "install", "maybe" ] >>= shouldBeSuccess
+            when installConsoleAndEffect do
+              spago [ "install", "--test-deps", "console", "effect" ] >>= shouldBeSuccess
             FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) "module Main where\nimport Prelude\nimport Data.Maybe\nimport Control.Alt\nmain = unit"
             -- get rid of "Compiling ..." messages and other compiler warnings
             spago [ "build" ] >>= shouldBeSuccess
 
         Spec.it "passed --pedantic-packages CLI flag" \{ spago, fixture } -> do
-          setupSrcTransitiveTests spago
+          setupSrcTransitiveTests spago true
           spago [ "build", "--pedantic-packages" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency.txt")
 
         Spec.it "package config has 'pedantic_packages: true'" \{ spago, fixture } -> do
-          setupSrcTransitiveTests spago
+          setupSrcTransitiveTests spago false
           addPedanticPackagesToSrc
           spago [ "build" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency.txt")
 
       Spec.describe "warns about unused dependencies when" do
         let
-          setupSrcUnusedDeps spago = do
+          setupSrcUnusedDeps spago installConsoleAndEffect = do
             spago [ "init", "--name", "7368613235362d2f444a2b4f56375435646a59726b53586548" ] >>= shouldBeSuccess
+            when installConsoleAndEffect do
+              spago [ "install", "--test-deps", "console", "effect" ] >>= shouldBeSuccess
             FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) "module Main where\nimport Prelude\nmain = unit"
             -- get rid of "Compiling ..." messages and other compiler warnings
             spago [ "build" ] >>= shouldBeSuccess
 
         Spec.it "passing --pedantic-packages CLI flag" \{ spago, fixture } -> do
-          setupSrcUnusedDeps spago
+          setupSrcUnusedDeps spago true
           spago [ "build", "--pedantic-packages" ] >>= shouldBeFailureErr (fixture "check-unused-dependency.txt")
 
         Spec.it "package config has 'pedantic_packages: true'" \{ spago, fixture } -> do
-          setupSrcUnusedDeps spago
+          setupSrcUnusedDeps spago false
```

Comment:
```suggestion
          setupSrcUnusedDeps spago true
```

I'd love to see a test that has these packages in test dependencies as well, to make sure we do not remove them unconditionally from the suggestion that we give out

Reply:
I didn't quite understand what was meant here, but I think latest commits addresses this point.

Reply:
Ah sorry, my comment was misguided. I meant to say that if we happen to have a package - let's say `effect` - both in the test and on the source dependencies, Spago should note both

Reply:
Ah, ok.
Hm... but that test doesn't pass :disappointed:  :smile:

Reply:
Actually, it does pass. I just needed to install the dependencies in `test` as well.

Reply:
See latest commit.